src/main/scala/org/apache/spark/shuffle/ucx/rpc/UcxDriverRpcEndpoint.scala

```diff
@@ -21,6 +22,15 @@ class UcxDriverRpcEndpoint(override val rpcEnv: RpcEnv) extends ThreadSafeRpcEnd
   private var executorToDpuAddress = HashMap.empty[Long, SerializableDirectBuffer]
   private var nvkvLock = new TrieMap[InetSocketAddress, Int]
   private var nvkvLockPendingQMap = new TrieMap[InetSocketAddress, mutable.ListBuffer[RpcEndpointRef]]
+  private var executerLocalIdMap = new TrieMap[InetSocketAddress, Int]
+  
+  private def getExecuterLocalId(key: InetSocketAddress): Int = {
+    val executerLocalId = executerLocalIdMap.replace(key, executerLocalIdMap(key)+1)
```

Comment:
If my understanding is correct, this function increases the "local id" each time it is called. Meaning, a **getter** function is not idempotent, but instead modifies data structs.

Reply:
Indeed name is confusing,`getAndSet` or other name implies that this function also change the value in the map will be better